lib/ferver/controller.rb

```diff
@@ -20,13 +20,30 @@ class Controller < Sinatra::Base
       halt 500, "Ferver: Directory '#{current_ferver_path}' not found."
     end
 
+    helpers do
+      def protected!
+        unless authorized?
+          response['WWW-Authenticate'] = %(Basic realm="Restricted Area")
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.